### PR TITLE
Feature/add report

### PR DESCRIPTION
**Pull request by Evgen-88**
Was implemented User Report & Exception Handler

### CODE REVIEW

File: src/main/java/com/itrex/konoplyanik/boardgamerent/controller/AccessoryController.java [hunk 1]
```diff
@@ -1,11 +1,8 @@
 package com.itrex.konoplyanik.boardgamerent.controller;
 
 import com.itrex.konoplyanik.boardgamerent.dto.AccessoryDTO;
-import com.itrex.konoplyanik.boardgamerent.exception.ServiceException;
 import com.itrex.konoplyanik.boardgamerent.service.AccessoryService;
 import lombok.RequiredArgsConstructor;
-import org.springframework.http.HttpStatus;
-import org.springframework.http.ResponseEntity;
 import org.springframework.security.access.prepost.PreAuthorize;
```
Comment (Ilia911) [inline src/main/java/com/itrex/konoplyanik/boardgamerent/controller/AccessoryController.java:6]: Delete asterisk imports

**Review by Ilia911 — COMMENTED** [review]
Reviewed.

File: src/main/java/com/itrex/konoplyanik/boardgamerent/controller/AccessoryController.java (hunk 1 above)
Comment (Evgen-88) [reply src/main/java/com/itrex/konoplyanik/boardgamerent/controller/AccessoryController.java:6]: thanks for advice

**Review by Ilia911 — APPROVED** [review]
Approved

File: src/main/java/com/itrex/konoplyanik/boardgamerent/controller/UserReportController.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    @GetMapping
+    public List<UserReportDTO> getUserReport(
+            @RequestParam("start") @DateTimeFormat(pattern = "yyyy-MM-dd") Date start,
+            @RequestParam("end") @DateTimeFormat(pattern = "yyyy-MM-dd") Date end,
+            @RequestParam(name = "pageNumber", defaultValue  = "1") Integer pageNumber,
+            @RequestParam(name = "pageSize", defaultValue  = "10") Integer pageSize,
+            @RequestParam(name = "parameter", required = false) String parameter,
+            @RequestParam(name = "direction", required = false) String direction
```
Comment (mbabrovich81) [inline src/main/java/com/itrex/konoplyanik/boardgamerent/controller/UserReportController.java]: I suggest to use special DTO for incoming parameters. Put all parameters in one object

File: src/main/java/com/itrex/konoplyanik/boardgamerent/controller/handler/ControllerExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+public class ControllerExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(MethodArgumentTypeMismatchException.class)
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ResponseEntity<?> handleMethodArgumentTypeMismatchException(
+            WebRequest request,
+            MethodArgumentTypeMismatchException ex
+    ) {
+        log.error("handleMethodArgumentTypeMismatchException {}\n", request.getDescription(false), ex);
+        return handleExceptionInternal(ex,"Method argument type mismatch", new HttpHeaders(), HttpStatus.BAD_REQUEST, request);
```
Comment (kolyago-vladislav) [inline src/main/java/com/itrex/konoplyanik/boardgamerent/controller/handler/ControllerExceptionHandler.java:28]: Very interesting way, I somehow didn’t realize that you can invoke the existing method of parent class

File: src/main/java/com/itrex/konoplyanik/boardgamerent/service/impl/UserReportServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import com.itrex.konoplyanik.boardgamerent.repository.UserReportRepository;
+import com.itrex.konoplyanik.boardgamerent.service.UserReportService;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.util.List;
+
+@Service
+@AllArgsConstructor
```
Comment (kolyago-vladislav) [inline src/main/java/com/itrex/konoplyanik/boardgamerent/service/impl/UserReportServiceImpl.java]: Use the annotation @RequiredArgsConstructor, Pasha advised me

File: src/main/java/com/itrex/konoplyanik/boardgamerent/controller/UserReportController.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+@RestController
+@RequestMapping("/reports")
+public class UserReportController {
+
+    private final UserReportService reportService;
+
+    public UserReportController(UserReportService reportService) {
+        this.reportService = reportService;
+    }
```
Comment (kolyago-vladislav) [inline src/main/java/com/itrex/konoplyanik/boardgamerent/controller/UserReportController.java]: Maybe it's worth using an annotation @RequiredArgsConstructor and removing the constructor?

Comment (Evgen-88) [reply src/main/java/com/itrex/konoplyanik/boardgamerent/controller/UserReportController.java]: Good idea. Thanks.

**Review by kolyago-vladislav — APPROVED** [review]

**Review by mbabrovich81 — APPROVED** [review]

**Pull request merged**